flink-metrics/flink-metrics-ganglia/pom.xml

```diff
@@ -68,20 +68,27 @@ under the License.
 	<build>
 		<plugins>
 			<plugin>
-				<artifactId>maven-assembly-plugin</artifactId>
-				<version>2.4</version>
-				<configuration>
-					<descriptorRefs>
-						<descriptorRef>jar-with-dependencies</descriptorRef>
-					</descriptorRefs>
-				</configuration>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<version>2.4.1</version>
 				<executions>
+					<!-- Run shade goal on package phase -->
 					<execution>
-						<id>make-assembly</id>
+						<id>shade-flink</id>
 						<phase>package</phase>
 						<goals>
-							<goal>single</goal>
+							<goal>shade</goal>
 						</goals>
+						<configuration combine.self="override">
+							<artifactSet>
+								<includes>
+									<include>org.apache.flink:flink-metrics-dropwizard</include>
+									<include>io.dropwizard.metrics:metrics-core</include>
+									<include>io.dropwizard.metrics:metrics-ganglia</include>
+									<include>info.ganglia.gmetric4j:gmetric4j</include>
```

Comment:
Sure that we don't need `org.acplt:oncrpc:jar:1.0.7:compile` which is a transitive dependency of `info.ganglia.gmetric4j` in the fat jar? Have you tried running ganglia on a cluster with this shaded jar?